src/components/DownloadResultButton.tsx

```diff
@@ -0,0 +1,36 @@
+import DownloadIcon from '@mui/icons-material/Download';
+import { Button, Tooltip, Typography } from '@mui/material';
+
+function DownloadResultButton({
+  identifier,
+  disabled,
+  handleClick,
+}: {
+  identifier: string;
+  disabled: boolean;
+  handleClick: () => void;
+}) {
+  const button = (
+    <Button
+      variant="contained"
+      startIcon={<DownloadIcon />}
+      onClick={handleClick}
+      disabled={disabled}
+    >
+      Download {identifier} Result
+    </Button>
+  );
+
+  return disabled ? (
```

Comment:
hhm, interesting. I see what you meant before. Let's go with this, I think that's clear and readable.